ouroboros-consensus/src/Ouroboros/Consensus/Crypto/Hash/Class.hs

```diff
@@ -24,17 +24,18 @@ import qualified Data.ByteString.Lazy as LB
 import           Data.List (foldl')
 import           Data.Proxy (Proxy (..))
 import           Data.String (IsString (..))
+import           Data.Typeable (Typeable)
 import           GHC.Generics (Generic)
 import           Numeric.Natural
 
 import           Ouroboros.Consensus.Util.Condense
 
-class HashAlgorithm h where
+class Typeable h => HashAlgorithm h where
```

Comment:
It would be nicer if all these `Typeable` changes were bundled in one preparatory commit :slightly_smiling_face: